t/porting/bidi-control.t

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/perl
+use v5.20.0;
+use warnings;
+
+use utf8 ();
+
+BEGIN {
+  @INC = '..' if -f '../TestInit.pm';
+  require './test.pl';
+}
+
+use TestInit qw(T); # T is chdir to the top level
+
+find_git_or_skip();
+
+my $cdup = `git rev-parse --show-cdup`;
+die "couldn't git-rev-parse --show-cdup\n" if $?;
+chomp $cdup;
+
+if (length $cdup) {
+  chdir $cdup or die "can't chdir to git root: $!";
+}
+
+my @files = `git ls-files`;
+die "couldn't git-ls-files\n" if $?;
+
+chomp @files;
+
+if (0) {
+  # Is this a useful filter?  Should we test every file?  Do we contain binary
+  # files, etc? -- rjbs, 2021-11-02
+  my %is_interesting = map {; $_ => 1 } qw( bat c h pl pm sh t xs );
+  @files = grep {; /\.([a-z]+)\z/ && $is_interesting{$1} } @files;
+}
```

Comment:
My gut reaction was that it shouldn't be useful - our source code repository should store store code. But I'm not aware there's ever been policy in this area, and on looking I see some obviously reasonable counterexamples such as test files for `IO::Compress`.

However we have 125 file extensions - 89 if you ignore `README.\w+` - and a high proportion of them appear to be text, with rather more than the short list above being code of some sort. Binary files are TTBOMK added rarely and changed far less often, so if none of the current crop happens to include Bidi controls I think it'd be far safer to leave the creation of an exception mechanism until the point we get a concrete example to base it around.

Reply:
Why not use file test operator -T.  It will return false on binary files.  Porting/makerel has code for creating an EBCDIC translation.  That works fine for deciding which files to use without any lists.